mcs/class/System.ServiceModel.Web/System.ServiceModel.Channels/WebMessageEncodingBindingElement.cs

```diff
@@ -146,7 +146,7 @@ public override T GetProperty<T> (BindingContext context)
 			return context.GetInnerProperty<T> ();
 		}
 
-#if !NET_2_1
+#if !(NET_2_1 || XAMMAC_4_5)
```

Comment:
Please use normalized, non nested boolean logic as much as possible.

!(a || b) -> !a && !b

The second form is easier to reason about.